parse/IntComplexValueRefParser.cpp

```diff
@@ -102,6 +102,30 @@ namespace parse {
                     ) [ _val = new_<ValueRef::ComplexVariable<int> >(_a, _b, _c, _f, _d, _e) ]
                 ;
 
+            ship_parts_owned
+                =   (
+                            tok.ShipPartsOwned_ [ _a = construct<std::string>(_1) ]
+                        >-( parse::label(Empire_token) >    int_value_ref [ _b = _1 ] )
+                        >-( parse::label(Name_token)   >    string_value_ref [ _d = _1 ] )
+                    ) [ _val = new_<ValueRef::ComplexVariable<int>>(_a, _b, _c, _f, _d, _e) ]
+                ;
+
+            ship_part_class_owned
+                = (
+                            tok.ShipPartClassOwned_ [ _a = construct<std::string>(_1) ]
+                        >-( parse::label(Empire_token) >    int_value_ref [ _b = _1 ] )
+                        >-( parse::label(Class_token)  >
```

Comment:
`as_string` seems to return `std::vector<char>` instead of `std::string`.  I'm not certain how to overcome that in this context, hence the use of int.